synapse/storage/data_stores/main/events.py

```diff
@@ -1929,32 +1923,95 @@ def get_all_updated_current_state_deltas_txn(txn):
             get_all_updated_current_state_deltas_txn,
         )
 
+    def _store_labels_txn(self, txn, event):
+        """Extract labels from an event and store them.
+
+        Args:
+            txn (LoggingTransaction): The transaction to execute.
```

Comment:
it's actually not a transaction, despite its confusing name, nor is it something we execute. (It does provide a method which will execute a database query, but that's something different).

It might be better to remain silent on the subject but... it's an object which implements the [cursor interface of the db-api](https://www.python.org/dev/peps/pep-0249/#cursor-objects). So you you could do:

```suggestion
            txn (LoggingTransaction): database cursor
```